src/sync/events/utils.ts

```diff
@@ -26,15 +27,21 @@ export const fetchBlock = async (chainId: number, blockNumber: number, force = f
         } else {
           const nativeTokenTransaction: es.ftTransfers.Event[] = [];
           const block = await getProvider(chainId).getBlockWithTransactions(blockNumber);
+
           // Create transactions array to store
           const transactions = (block.transactions as ITransactionResponse[]).map(
             (tx: ITransactionResponse) => {
               // eslint-disable-next-line @typescript-eslint/no-explicit-any
               const rawTx = tx.raw as any;
 
               const gasPrice = tx.gasPrice?.toString();
-              const gasUsed = rawTx?.gas ? bn(rawTx.gas).toString() : undefined;
+              const gasLimit = tx.gasLimit?.toString();
+
+              let gasUsed = rawTx?.gas ? bn(rawTx.gas).toString() : undefined;
```

Comment:
I've found this to be always undefined. As rawTx doesn't exist. So, on next line I'm setting **gasUsed** as `tx.gasLimit`